sdk-lib/src/main/java/cash/z/ecc/android/sdk/block/processor/CompactBlockProcessor.kt

```diff
@@ -532,10 +847,10 @@ class CompactBlockProcessor internal constructor(
         if (failedUtxoFetches < 9) { // there are 3 attempts per block
             @Suppress("TooGenericExceptionCaught")
             try {
-                retryUpTo(UTXO_FETCH_RETRIES) {
+                retryUpToAndThrow(UTXO_FETCH_RETRIES) {
                     val tAddresses = backend.listTransparentReceivers(account)
 
-                    downloader.lightWalletClient.fetchUtxos(
+                    downloader.fetchUtxos(
                         tAddresses,
                         BlockHeightUnsafe.from(startHeight)
                     ).onEach { response ->
```

Comment:
Nitpick: there's a grammar error in the second sentence of the error message on line 903 ("It seems to unavailable on lightwalletd.").